generic: mtk_eth_soc: reduce driver memory usage

romanovj opened the pull request:
1. Import pending patch to fix ramips/mt7621 64MB devices,
which become broken after https://github.com/torvalds/linux/commit/c57e558194430d10d5e5f4acd8a8655b68dade13

https://lore.kernel.org/all/20241015110940.63702-3-nbd@nbd.name/

Older chipsets are typically deployed in systems that are more memory
constrained and don't actually need the larger rings to handle received
packets.

In order to reduce wasted memory set the ring size based on the SoC
to the following values:
- 2048 on MT7988
- 1024 on MT7986
- 512 (previous value) on everything else, except:
- 256 on RT5350 (the oldest supported chipset)

Fixes https://github.com/openwrt/openwrt/issues/17401
Fixes https://github.com/openwrt/openwrt/issues/16687
Fixes https://github.com/openwrt/openwrt/issues/16688

2. Further reduce memory usage by disabling CONFIG_PAGE_POOL_STATS
by default. Like it should be by default.

Signed-off-by: Danila Romanov <pervokur@gmail.com>

ptpt52 commented:
nice!

danpawlik commented:
@romanovj sorry for that question, but I see change for MT7988, where in that patch https://github.com/torvalds/linux/commit/c57e558194430d10d5e5f4acd8a8655b68dade13 it is. Should it be not other patch?

dangowrt commented:
Imho we should import all 4 patches of the series and maybe also help by re-sending it with a cover letter.
https://patchwork.kernel.org/project/netdevbpf/list/?series=899258&state=*

Generally everything there looks fine. nit: I'd give MT7981 the same value as MT7986 as Ethernet performance of MT7981 should be very similar to MT7986.

@nbd168 Will you resend or should I take care of it?

erantake committed:
generic: mtk_eth_soc: reduce driver memory usage

1. Import pending patch to fix ramips/mt7621 64MB targets.

2. Do not enable CONFIG_PAGE_POOL_STATS by default.

Signed-off-by: Danila Romanov <pervokur@gmail.com>

LuisMitaHL commented:
Is there anything else to try? I think this PR is important and should be in the last rc of 24.10, which I understand could be tagged on Wednesday.

hauke commented:
@nbd168  Could you please have a look at this PR

nbd168 commented:
Applied with some minor patch cleanup (no code changes), thanks.
